packages/terra-framework-docs/src/terra-dev-site/doc/worklist-data-grid/Examples.2/PinnedColumnsWithRowSelection.jsx

```diff
@@ -55,33 +55,15 @@ const RowSelection = () => {
   const rowSelectionModeRef = useRef();
   const rowHeaderIndex = 0;
   const { cols, rows } = gridDataJSON;
-  const [selectedRows, setSelectedRows] = useState([]);
+  const [rowData, setRowData] = useState(rows);
   const [hasSelectableRows, setHasSelectableRows] = useState(false);
 
-  const determineSelectedRows = useCallback((allRowsSelected, userSelectedRow) => {
-    if (!userSelectedRow) {
-      return [];
-    }
-
-    let remainingSelectedRow = [];
-    if (allRowsSelected) {
-      remainingSelectedRow = userSelectedRow;
-    } else if (selectedRows.includes(userSelectedRow[0])) {
-      // Row Deselected so remove this rowId.
-      remainingSelectedRow = selectedRows.filter(row => (row !== userSelectedRow[0]));
-    } else {
-      // Row Selected so add this rowId.
-      remainingSelectedRow = remainingSelectedRow.concat(selectedRows);
-      remainingSelectedRow.push(userSelectedRow[0]);
-    }
-    return remainingSelectedRow;
-  }, [selectedRows]);
-
   const clearRowSelection = useCallback(() => {
-    // eslint-disable-next-line no-param-reassign
-    rows.forEach(r => { if (r.isSelected) { r.isSelected = false; } });
-    setSelectedRows([]);
-  }, [rows]);
+    const newRowData = [...rowData];
+    // eslint-disable-next-line no-return-assign, no-param-reassign
+    newRowData.forEach(row => (row.isSelected = false));
+    setRowData(newRowData);
+  }, [rowData]);
```

Comment:
Curious, why not just use `map` here? It should be safe because `newRowData` is not mutating `rowData`. It could be:
```suggestion
    const newRowData = rowData.map(row => ({
      ...row,
      isSelected: false
    });
    setRowData(newRowData);
  }, [rowData]);
```